src/vcpkg/binarycaching.cpp

```diff
@@ -1101,36 +1111,41 @@ namespace
                 return;
             }
 
-            int64_t cache_size;
-            {
-                auto archive = fs.open_for_read(tmp_archive_path, VCPKG_LINE_INFO);
-                archive.try_seek_to(0, SEEK_END);
-                cache_size = archive.tell();
-            }
+            auto cache_size = fs.file_size(tmp_archive_path, VCPKG_LINE_INFO);
 
             size_t upload_count = 0;
             if (!m_write_url.empty())
             {
-                if (auto cacheId = reserve_cache_entry(abi, cache_size))
+                if (auto cacheId = reserve_cache_entry(spec.name(), abi, cache_size))
                 {
-                    std::vector<std::string> headers{
+                    std::vector<std::string> custom_headers{
```

Comment:
As a note: any time you create a `std::vector<std::string>`, that creates copies all the strings passed in (and not moved). This means that it isn't necessarily useful for those other strings to be `std::string` -- it can be more efficient for them to be `StringLiteral`s that you call `.to_string()` on.